anaconda upload failing with 'file does not exist'

`anaconda upload` from both CircleCI and TravisCI is failing for the new release with errors like this [[ref](https://circleci.com/gh/conda-forge/lal-feedstock/22), [ref](https://travis-ci.org/conda-forge/lal-feedstock/builds/433379714)]:

```
[ERROR] File "/home/conda/feedstock_root/build_artifacts/linux-64/lal-6.19.0-hee24be8_0.tar.bz2" does not exist
Traceback (most recent call last):
  File "/opt/conda/bin/upload_package", line 11, in <module>
    sys.exit(upload_package())
  File "/opt/conda/lib/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/opt/conda/lib/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/opt/conda/lib/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/conda/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/opt/conda/lib/python3.6/site-packages/conda_forge_ci_setup/build_utils.py", line 82, in upload_package
    upload_or_check(recipe_root, owner, channel, [config_file])
  File "/opt/conda/lib/python3.6/site-packages/conda_forge_ci_setup/upload_or_check_non_existence.py", line 135, in upload_or_check
    upload(token_fn, path, owner, channel)
  File "/opt/conda/lib/python3.6/site-packages/conda_forge_ci_setup/upload_or_check_non_existence.py", line 65, in upload
    env=os.environ)
  File "/opt/conda/lib/python3.6/subprocess.py", line 291, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['anaconda', '--quiet', '-t', '/tmp/tmptxtgcqgp/binstar.token', 'upload', '/home/conda/feedstock_root/build_artifacts/linux-64/lal-6.19.0-hee24be8_0.tar.bz2', '--user=conda-forge', '--channel=main']' returned non-zero exit status 1.
Exited with code 1
```

cc: @conda-forge/core - help is greatly appreciated.

--- a/recipe/build.sh
+++ b/recipe/build.sh
@@ -9,7 +9,7 @@
 set -e
 
 ./configure \
-	--prefix="$(pwd)/build" \
+	--prefix="${PREFIX}" \
 	--enable-swig-iface \
 	--disable-swig-octave \
 	--disable-swig-python \
@@ -18,4 +18,3 @@ set -e
 	--enable-silent-rules
 make -j ${CPU_COUNT}
 make -j ${CPU_COUNT} check
-make -j ${CPU_COUNT} install

--- a/recipe/build.sh
+++ b/recipe/build.sh
@@ -1,13 +1,16 @@
 #!/bin/sh
+#
+# Configure, build, and test a LALSuite subpackage (e.g. `lal`), including
+# the SWIG interface files, but without any actual language bindings
 
-[ -f ./00boot ] && ./00boot
 ./configure \
     --prefix=$PREFIX \
-    --enable-swig-python \
+    --enable-swig \
+    --enable-swig-iface \
     --disable-swig-octave \
+    --disable-swig-python \
+    --disable-python \
     --disable-doxygen \
     --disable-gcc-flags \
     --enable-silent-rules
 make -j ${CPU_COUNT}
-make -j ${CPU_COUNT} check
-make -j ${CPU_COUNT} install

--- a/recipe/build.sh
+++ b/recipe/build.sh
@@ -6,13 +6,11 @@
 set -e
 
 ./configure \
-    --prefix=$PREFIX \
-    --enable-swig \
-    --enable-swig-iface \
-    --disable-swig-octave \
-    --disable-swig-python \
-    --disable-python \
-    --disable-doxygen \
-    --disable-gcc-flags \
-    --enable-silent-rules
+	--prefix=$PREFIX \
+	--enable-swig-iface \
+	--disable-swig-octave \
+	--disable-swig-python \
+	--disable-python \
+	--disable-gcc-flags \
+	--enable-silent-rules
 make -j ${CPU_COUNT}

--- a/recipe/build.sh
+++ b/recipe/build.sh
@@ -3,6 +3,8 @@
 # Configure, build, and test a LALSuite subpackage (e.g. `lal`), including
 # the SWIG interface files, but without any actual language bindings
 
+set -e
+
 ./configure \
     --prefix=$PREFIX \
     --enable-swig \

--- a/recipe/build.sh
+++ b/recipe/build.sh
@@ -1,12 +1,15 @@
-#!/bin/sh
+#!/bin/bash
 #
 # Configure, build, and test a LALSuite subpackage (e.g. `lal`), including
 # the SWIG interface files, but without any actual language bindings
+#
+# This script installs to a dummy location, which is then copied into
+# the host env by install-c.sh
 
 set -e
 
 ./configure \
-	--prefix=$PREFIX \
+	--prefix="$(pwd)/build" \
 	--enable-swig-iface \
 	--disable-swig-octave \
 	--disable-swig-python \
@@ -14,3 +17,5 @@ set -e
 	--disable-gcc-flags \
 	--enable-silent-rules
 make -j ${CPU_COUNT}
+make -j ${CPU_COUNT} check
+make -j ${CPU_COUNT} install

--- a/recipe/install-c.sh
+++ b/recipe/install-c.sh
@@ -1,0 +1,5 @@
+#!/bin/bash
+
+# top level build has done everything, so just test it, and install it
+make -j ${CPU_COUNT} check
+make install

--- a/recipe/install-c.sh
+++ b/recipe/install-c.sh
@@ -2,11 +2,5 @@
 
 set -e
 
-. activate "${PREFIX}"
-pushd ${SRC_DIR}
-
-# top level build has done everything, so just test it, and install it
-make -j ${CPU_COUNT} check
-make install
-
-popd
+# copy everything from fake build to actual install
+cp -vr ${SRC_DIR}/build/* ${PREFIX}/

--- a/recipe/install-c.sh
+++ b/recipe/install-c.sh
@@ -1,5 +1,7 @@
 #!/bin/bash
 
+set -e
+
 # top level build has done everything, so just test it, and install it
 make -j ${CPU_COUNT} check
 make install

--- a/recipe/install-c.sh
+++ b/recipe/install-c.sh
@@ -2,5 +2,5 @@
 
 set -e
 
-# copy everything from fake build to actual install
-cp -vr ${SRC_DIR}/build/* ${PREFIX}/
+source activate "${PREFIX}"
+make install

--- a/recipe/install-c.sh
+++ b/recipe/install-c.sh
@@ -2,6 +2,11 @@
 
 set -e
 
+. activate "${PREFIX}"
+pushd ${SRC_DIR}
+
 # top level build has done everything, so just test it, and install it
 make -j ${CPU_COUNT} check
 make install
+
+popd

--- a/recipe/install-python.sh
+++ b/recipe/install-python.sh
@@ -2,6 +2,9 @@
 
 set -e
 
+. activate "${PREFIX}"
+pushd ${SRC_DIR}
+
 # configure only python bindings and pure-python extras
 ./configure \
 	--prefix=$PREFIX \
@@ -21,3 +24,5 @@ make -j ${CPU_COUNT} -C test/python check
 # install
 make -j ${CPU_COUNT} -C swig install-exec-am  # swig bindings
 make -j ${CPU_COUNT} -C python install  # pure-python extras
+
+popd

--- a/recipe/install-python.sh
+++ b/recipe/install-python.sh
@@ -2,7 +2,6 @@
 
 set -e
 
-export PATH="${CONDA_DEFAULT_ENV}/bin:${PATH}"
 . activate "${PREFIX}"
 
 pushd ${SRC_DIR}

--- a/recipe/install-python.sh
+++ b/recipe/install-python.sh
@@ -9,7 +9,7 @@ pushd ${SRC_DIR}
 # configure only python bindings and pure-python extras
 ./configure \
 	--prefix=$PREFIX \
-	--enable-swig-iface \
+	--disable-swig-iface \
 	--enable-swig-python \
 	--enable-python \
 	--disable-doxygen \

--- a/recipe/install-python.sh
+++ b/recipe/install-python.sh
@@ -1,5 +1,7 @@
 #!/bin/bash
 
+set -e
+
 # build only python bindings and pure-python extras
 ./configure \
 	--prefix=$PREFIX \

--- a/recipe/install-python.sh
+++ b/recipe/install-python.sh
@@ -2,7 +2,9 @@
 
 set -e
 
+export PATH="${CONDA_DEFAULT_ENV}/bin:${PATH}"
 . activate "${PREFIX}"
+
 pushd ${SRC_DIR}
 
 # configure only python bindings and pure-python extras
@@ -16,7 +18,8 @@ pushd ${SRC_DIR}
 	--enable-silent-rules
 
 # build
-make -j ${CPU_COUNT}
+make -j ${CPU_COUNT} -C swig
+make -j ${CPU_COUNT} -C python
 
 # test
 make -j ${CPU_COUNT} -C test/python check

--- a/recipe/install-python.sh
+++ b/recipe/install-python.sh
@@ -2,15 +2,17 @@
 
 set -e
 
-# build only python bindings and pure-python extras
+# configure only python bindings and pure-python extras
 ./configure \
 	--prefix=$PREFIX \
-	--disable-doxygen \
-	--disable-swig \
+	--enable-swig-iface \
 	--enable-swig-python \
 	--enable-python \
+	--disable-doxygen \
 	--disable-gcc-flags \
 	--enable-silent-rules
+
+# build
 make -j ${CPU_COUNT}
 
 # test

--- a/recipe/install-python.sh
+++ b/recipe/install-python.sh
@@ -15,7 +15,7 @@ pushd ${SRC_DIR}
 	--enable-python \
 	--disable-doxygen \
 	--disable-gcc-flags \
-	--enable-silent-rules
+	--enable-silent-rules || { cat config.log; exit 1; }
 
 # build
 make -j ${CPU_COUNT} -C swig

--- a/recipe/install-python.sh
+++ b/recipe/install-python.sh
@@ -1,0 +1,19 @@
+#!/bin/bash
+
+# build only python bindings and pure-python extras
+./configure \
+	--prefix=$PREFIX \
+	--disable-doxygen \
+	--disable-swig \
+	--enable-swig-python \
+	--enable-python \
+	--disable-gcc-flags \
+	--enable-silent-rules
+make -j ${CPU_COUNT}
+
+# test
+make -j ${CPU_COUNT} -C test/python check
+
+# install
+make -j ${CPU_COUNT} -C swig install-exec-am  # swig bindings
+make -j ${CPU_COUNT} -C python install  # pure-python extras

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -86,7 +86,7 @@ outputs:
         - python
         - {{ pin_compatible('numpy') }}
         - six
-        - dateutil
+        - python-dateutil
         - ligo-segments
         - lscsoft-glue
         - scipy

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -76,6 +76,10 @@ outputs:
         - make
         - {{ compiler('c') }}
       host:
+        - gsl
+        - zlib
+        - fftw
+        - hdf5
         - {{ pin_subpackage('lal', exact=True) }}
         - swig >=3.0.7
         - python

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -12,7 +12,7 @@ source:
   sha256: {{ sha256 }}
 
 build:
-  number: 0
+  number: 1
   skip: true  # [win]
 
 requirements:
@@ -21,20 +21,13 @@ requirements:
     - make
     - {{ compiler('c') }}
     - bc
-  host:
-    - swig >=3.0.7
     - blas 1.1 openblas
     - openblas
     - gsl
     - zlib
     - fftw
     - hdf5
-    - python
-    - numpy
-    - six
-    - ligo-segments
-    - lscsoft-glue >=1.54.1
-    - scipy
+    - swig >=3.0.7
 
 test:
   imports:
@@ -44,7 +37,14 @@ test:
 
 outputs:
   - name: lal
+    script: install-c.sh
     requirements:
+      build:
+        - pkg-config
+        - make
+        - {{ compiler('c') }}
+        - bc
+        - swig >=3.0.7
       host:
         - blas 1.1 openblas
         - openblas
@@ -59,13 +59,6 @@ outputs:
         - zlib
         - fftw
         - hdf5
-    files:
-      - bin/lal_version
-      - bin/lal_simd_detect
-      - etc/lal-user-env.*
-      - include/lal/
-      - lib/liblal*
-      - lib/pkgconfig/lal*.pc
     test:
       commands:
         - lal_version --verbose
@@ -83,25 +76,35 @@ outputs:
         This package contains the shared-object libraries and python bindings
         needed to run applications that use the LAL library.
   - name: python-lal
+    script: install-python.sh
     requirements:
+      build:
+        - make
+        - {{ compiler('c') }}
       host:
         - {{ pin_subpackage('lal', exact=True) }}
         - swig >=3.0.7
-        - python
-        - numpy
-      run:
-        - {{ pin_subpackage('lal', exact=True) }}
         - python
         - numpy
         - six
         - ligo-segments
         - lscsoft-glue
         - scipy
-    files:
-      - lib/python*/site-packages/lal/**
+      run:
+        - {{ pin_subpackage('lal', exact=True) }}
+        - python
+        - {{ pin_compatible('numpy') }}
+        - six
+        - dateutil
+        - ligo-segments
+        - lscsoft-glue
+        - scipy
     test:
       imports:
         - lal
+        - lal.gpstime
+        - lal.rate
+        - lal.series
         - lal.utils
     about:
       home: https://wiki.ligo.org/DASWG/LALSuite

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -34,9 +34,10 @@ outputs:
   - name: lal
     script: install-c.sh
     requirements:
-      build:
-        - {{ compiler('c') }}
       host:
+        - {{ compiler('c') }}
+        - make
+        - swig >=3.0.7
         - blas 1.1 openblas
         - openblas
         - gsl

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -21,6 +21,7 @@ requirements:
     - make
     - {{ compiler('c') }}
     - bc
+  host:
     - blas 1.1 openblas
     - openblas
     - gsl
@@ -34,11 +35,7 @@ outputs:
     script: install-c.sh
     requirements:
       build:
-        - pkg-config
-        - make
         - {{ compiler('c') }}
-        - bc
-        - swig >=3.0.7
       host:
         - blas 1.1 openblas
         - openblas
@@ -73,15 +70,12 @@ outputs:
     script: install-python.sh
     requirements:
       build:
-        - make
         - {{ compiler('c') }}
+        - make
+        - pkg-config
       host:
-        - gsl
-        - zlib
-        - fftw
-        - hdf5
-        - {{ pin_subpackage('lal', exact=True) }}
         - swig >=3.0.7
+        - {{ pin_subpackage('lal', exact=True) }}
         - python
         - numpy
         - six

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -29,12 +29,6 @@ requirements:
     - hdf5
     - swig >=3.0.7
 
-test:
-  imports:
-    - lal
-  commands:
-    - lal_version --help
-
 outputs:
   - name: lal
     script: install-c.sh

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -69,11 +69,10 @@ outputs:
   - name: python-lal
     script: install-python.sh
     requirements:
-      build:
+      host:
         - {{ compiler('c') }}
         - make
         - pkg-config
-      host:
         - swig >=3.0.7
         - {{ pin_subpackage('lal', exact=True) }}
         - python

